cg/meta/workflow/nf_analysis.py

```diff
@@ -871,17 +872,7 @@ def get_genome_build(self, case_id: str) -> GenomeVersion:
         """Return reference genome version for a case.
         Raises CgError if this information is missing or inconsistent for the samples linked to a case.
         """
-        reference_genome: set[str] = {
-            sample.reference_genome
-            for sample in self.status_db.get_samples_by_case_id(case_id=case_id)
-        }
-        if len(reference_genome) == 1:
-            return reference_genome.pop()
-        if len(reference_genome) > 1:
-            raise CgError(
-                f"Samples linked to case {case_id} have different reference genome versions set"
-            )
-        raise CgError(f"No reference genome specified for case {case_id}")
+        return get_genome_build(case_id=case_id, status_db=self.status_db)
```

Comment:
I think you can separate the database interaction to this function and then the consensus logic to the new function
```suggestion
        case = self.status_db.get_case_by_internal_id(case_id)
        return get_genome_build(case)
```